js/colourpicker.js

```diff
@@ -608,101 +1042,117 @@
-
-    // Converts a hex string to an HSB object
     function hex2hsb(hex) {
       var hsb = rgb2hsb(hex2rgb(hex));
       if( hsb.s === 0 ) hsb.h = 360;
       return hsb;
     }
 
-    // Converts an RGB object to an HSB object
-    function rgb2hsb(rgb) {
-      var hsb = { h: 0, s: 0, b: 0 };
-      var min = Math.min(rgb.r, rgb.g, rgb.b);
-      var max = Math.max(rgb.r, rgb.g, rgb.b);
-      var delta = max - min;
-      hsb.b = max;
-      hsb.s = max !== 0 ? 255 * delta / max : 0;
-      if( hsb.s !== 0 ) {
-        if( rgb.r === max ) {
-          hsb.h = (rgb.g - rgb.b) / delta;
-        } else if( rgb.g === max ) {
-          hsb.h = 2 + (rgb.b - rgb.r) / delta;
-        } else {
-          hsb.h = 4 + (rgb.r - rgb.g) / delta;
-        }
-      } else {
-        hsb.h = -1;
-      }
-      hsb.h *= 60;
-      if( hsb.h < 0 ) {
-        hsb.h += 360;
-      }
-      hsb.s *= 100/255;
-      hsb.b *= 100/255;
-      return hsb;
+    function hex2hsl(hex) {
+      return rgb2hsl(hex2rgb(hex));
     }
 
-    // Converts a hex string to an RGB object
     function hex2rgb(hex) {
-      hex = parseInt(((hex.indexOf('#') > -1) ? hex.substring(1) : hex), 16);
-      return {
-        /* jshint ignore:start */
-        r: hex >> 16,
-        g: (hex & 0x00FF00) >> 8,
-        b: (hex & 0x0000FF)
-        /* jshint ignore:end */
-      };
+      hex = correctedHex(hex);
+      return str2rgb(hex);
+    } 
+
+    function hex2name(hex) {
+      hex = correctedHex(hex);
+      hex8 = correctedHex(hex,8);
+      if (!hex || !hex8) return '';
+      //if alpha = 0 return 'transparent' no matter what the other values are
+      if (hex8.slice(-2) === "00") return 'transparent';
+      //otherwise look it up using hex8, or return the hex code itself if need be.
+      if (hex8 in colsToNames) return colsToNames[hex8]; else return hex;
+    }
+
+    function hsl2rgb(hsl) {
+      hsl = correctedHSL(hsl);
+      var r,g,b,a
+      h = keepWithin(hsl.h,0,360)/360;
+      s = keepWithin(hsl.s,0,100)/100;
+      l = keepWithin(hsl.l,0,100)/100;
+      a = (hsl.a===false) ? false : keepWithin(hsl.a,0,1);
+      function hue2rgb(p, q, t) {
+            if(t < 0) t += 1;
+            if(t > 1) t -= 1;
+            if(t < 1/6) return p + (q - p) * 6 * t;
+            if(t < 1/2) return q;
+            if(t < 2/3) return p + (q - p) * (2/3 - t) * 6;
+            return p;
+      }
+
+      if(s === 0) {
+          r = g = b = l; // achromatic
+      } else {
+          var q = l < 0.5 ? l * (1 + s) : l + s - l * s;
+          var p = 2 * l - q;
+          r = hue2rgb(p, q, h + 1/3);
+          g = hue2rgb(p, q, h);
+          b = hue2rgb(p, q, h - 1/3);
+      }
+      return {r: Math.round(r*255), g: Math.round(g*255), b: Math.round(b*255), a: (a ? roundTo(a,4) : a)};
+    }
+    function hsl2hex(hsl,force) {
+      return rgb2hex(hsl2rgb(hsl),force);
+    }
+    function hsl2hsb(hsl) {
+      return rgb2hsb(hsl2rgb);
+    }
+    function hsl2name(hsl) {
```

Comment:
Most of this code came from TinyColor (again, I should credit that, oops, my credit must have gotten lost). I did add some sanity checks and such. If you look at the functions carefully you'l l see that most of them basically DO do that - they convert everything internally back and forth to rgb. And in practice, all of my calls to it bascially do that too, since the internal colour is stored as rgb. I'll try to remove the unnecessary ones.